crates/sui-source-validation-service/src/lib.rs

```diff
@@ -414,18 +417,40 @@ async fn api_route(
     let symbol = Symbol::from(module);
     let Ok(address) = AccountAddress::from_hex_literal(&address) else {
 	let error = format!("Invalid hex address {address}");
-	return (StatusCode::BAD_REQUEST, headers, Json(ErrorResponse { error }).into_response())
-    };
-    let Some(SourceInfo {source : Some(source), ..}) = app_state.sources.get(&network).and_then(|l| l.get(&(address, symbol))) else {
-	let error = format!("No source found for {symbol} at address {address} on network {network}");
-	return (StatusCode::NOT_FOUND, headers, Json(ErrorResponse { error }).into_response())
+	return (
+	    StatusCode::BAD_REQUEST,
+	    headers,
+	    Json(ErrorResponse { error }).into_response()
+	)
     };
-    (
-        StatusCode::OK,
-        headers,
-        Json(SourceResponse {
-            source: source.to_owned(),
-        })
-        .into_response(),
-    )
+
+    let source_result = app_state
+        .sources
+        .get(&network)
+        .and_then(|l| l.get(&(address, symbol)));
+    if let Some(SourceInfo {
```

Comment:
`let ... else` here as well?

Reply:
so funnily enough `rustfmt` doesn't have a problem formatting `if let ... { } else { }` with braces, only `let ... else`. I wanted to transform this to something like `let x = expr?;` but since web responses aren't `Result` type it doesn't quite work, so opted for `if let ...`